File: src/main/scala/assignment3/e0/ParticleSystemNoGUI.java
package assignment3.e0;

/**
 * PCD 2018-2019 - Assignment #01
 * 
 * Particle system version for profiling purposes (no GUI)
 * 
 * @author aricci
 *
 */
public class ParticleSystemNoGUI {
	public static void main(String[] args) {
		
		int nParticles = 1000;
		int nSteps = 1000;
		double dt = 0.01;
		
		World world = new World(dt);
		Flag stopFlag = new Flag();
		ParticleMaster master = new ParticleMaster(world, stopFlag, nParticles, nSteps);
		master.start();
	}

}

File: src/main/scala/assignment3/e0/ParticleSystemNoGUI.scala
package assignment3.e0

import akka.actor.ActorSystem
import assignment3.e0.actors.ActorActions.Start
import assignment3.e0.actors.{ActorActions, ControllerActor}


object ParticleSystemNoGUI {

  def main(args: Array[String]): Unit = {
    val nParticles = 1000
    val nSteps: Int = 1000
    val dt = 0.01

    val world = new World(dt)

    val system = ActorSystem("ParticleSystem")
    val controllerActor = system.actorOf(ControllerActor.props(world, nSteps), "ControllerActor")

    controllerActor ! Start(nParticles)
  }

}